src/sentry_envelope.c

```diff
@@ -459,49 +500,85 @@ sentry__envelope_for_each_request(const sentry_envelope_t *envelope,
         endpoint_type = ENDPOINT_TYPE_MINIDUMP;
     }
 
-    if (attachment_count == 0) {
-        return;
-    }
-
-    char boundary[50];
-    gen_boundary(boundary);
+    if (other_count > 0) {
+        sentry_stringbuilder_t sb;
+        sentry__stringbuilder_init(&sb);
+        sentry__envelope_serialize_headers_into_stringbuilder(envelope, &sb);
+        for (size_t i = 0; i < other_count; i++) {
+            sentry__envelope_serialize_item_into_stringbuilder(other[i], &sb);
+        }
+        size_t body_len = sentry__stringbuilder_len(&sb);
+        char *body = sentry__stringbuilder_into_string(&sb);
+        req = prepare_http_request(&event_id, endpoint_type,
+            "application/x-sentry-envelope", body, body_len, true);
```

Comment:
hm, since we already require a sentry which can deal with envelopes, I was considering to massively simplify this (throwing out all the multipart handling), and basically make `envelope 1 <-> 1 http request` a 1:1 relation. but that is for the future.

Reply:
We can't do that unfortunately because of different rate limits.

Reply:
Just so I understand the intention correctly…
* In the normal case, we want to drop events from `capture_event` on rate-limit, right?
* Also, with the changes to `capture_event`, we attach the session to the captured envelope, and want to send it as different http requests for rate limiting purposes…
* Then, on normal shutdown, we close the session and send it as its own envelope.

I wonder why we don’t just create a separate envelope for it?
Also since you mentioned raw envelopes, how should we proceed with that? I would personally prefer to still have a 1:1 relation between event->envelope->http request.

In any case, I think the rate limiting itself needs some work in the transports.